lib/IRGen/GenObjC.h

```diff
@@ -105,58 +105,51 @@ namespace irgen {
   llvm::Value *emitObjCAutoreleaseReturnValue(IRGenFunction &IGF,
                                               llvm::Value *value);
 
+  struct ObjCMethodRep {
+    struct DescriptorParts {
+      llvm::Constant *selectorRef;
+      llvm::Constant *typeEncoding;
+      llvm::Constant *impl;
+    };
+
+    DescriptorParts descriptor;
+    SILFunction *silFunction;
+  };
```

Comment:
I don't think having both levels here is buying you much; please flatten to a single struct.  And I'd just call it `ObjCMethodDescriptor`.

Please initialize all the pointers to `nullptr`.

Reply:
Done.